### PR TITLE
Using controller-utils to get workloads

### PR DESCRIPTION
**Pull request by jdesouza**
This PR fixes #

## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

### What's the goal of this PR?
Improve how polaris retrieves the workloads 
### What changes did you make?
Using controller-utils to retrieve controllers

### CODE REVIEW

File: pkg/kube/resources.go [hunk 1]
```diff
@@ -358,17 +352,28 @@ func CreateResourceProviderFromAPI(ctx context.Context, kube kubernetes.Interfac
...
+		workloadObj, err := NewGenericResourceFromUnstructured(topController, nil)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse workload %v: %w", workload, err)
+		}
+		workloads = append(workloads, workloadObj)
 	}
+
 	// resources loaded from custom checks can also contain controllers and thus would be added twice to the provider
-	kubernetesResources = deduplicateControllers(append(kubernetesResources, controllers...))
+	kubernetesResources = deduplicateControllers(append(kubernetesResources, workloads...))
```
Comment (rbren) [inline pkg/kube/resources.go:376]: I think we can remove this call--controller-utils should do the deduping for us

File: pkg/kube/resources.go [hunk 2]
```diff
@@ -358,17 +352,28 @@ func CreateResourceProviderFromAPI(ctx context.Context, kube kubernetes.Interfac
 			kubernetesResources = append(kubernetesResources, res)
 		}
 	}
-
-	objectCache := map[string]unstructured.Unstructured{}
-
 	logrus.Info("Loading controllers")
-	controllers, err := LoadControllers(ctx, pods.Items, dynamic, restMapper, objectCache)
```
Comment (rbren) [inline pkg/kube/resources.go:365]: I'm guessing this function is dead code now--can we delete?

Comment (jdesouza) [reply pkg/kube/resources.go:365]: done

File: pkg/kube/resources.go (hunk 1 above)
Comment (jdesouza) [reply pkg/kube/resources.go]: done

**Review by rbren — APPROVED** [review]

**Pull request merged**